src/Compilers/CSharp/Portable/Symbols/Source/ParameterHelpers.cs

```diff
@@ -79,6 +79,14 @@ internal static class ParameterHelpers
                     diagnostics.Add(ErrorCode.ERR_IllegalRefParam, refnessKeyword.GetLocation());
                 }
 
+                if (parameterSyntax.ExclamationToken.Kind() == SyntaxKind.ExclamationToken)
+                {
+                    if (owner.IsAbstract || owner.IsImplementableInterfaceMember() || owner.IsPartialDefinition() || owner.IsExtern)
```

Comment:
>owner.IsImplementableInterfaceMember() [](start = 44, length = 38)

I'm not sure that this will work in the face of Default Interface Members. Please add some tests for that scenario.